lib/image.js

```diff
@@ -46,20 +43,22 @@ Object.defineProperty(Image.prototype, 'src', {
             throw err
           }
         }
-
-        if (!get) get = require('simple-get')
-
-        get.concat({
-          url: val,
+        const request =  fetch(url = val, data = {
+          method: "POST",
           headers: { 'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.0.0 Safari/537.36' }
-        }, (err, res, data) => {
-          if (err) return onerror(err)
+        })
 
+        request.then((res) => {
           if (res.statusCode < 200 || res.statusCode >= 300) {
             return onerror(new Error(`Server responded with ${res.statusCode}`))
           }
-
-          setSource(this, data)
+          return res.arrayBuffer()
+        })
+        .then((data) => {
+          return setSource(this, data)
+        })
+        .catch((err) => {
+          return onerror(err)
```

Comment:
Same here.

```suggestion
          onerror(err)
```

Also, the behaviour of `onerror` is that it throws if a handler hasn't been set. Before this change, that thrown was delegated to the [global error handler](https://nodejs.org/docs/latest-v20.x/api/process.html#event-uncaughtexception).

Now it will instead be eaten by this promise, which I don't think is desirable.

I think that `util.callbackify` or `process.nextTick` would be best used to restore the old behaviour 🤔

Reply:
=> https://github.com/Automattic/node-canvas/pull/2309/commits/93cfd2f012cf40bbf0d7a7f88f60b9c3659106a2

I'm not sure if the changes I made are correct...